### PR TITLE
Fix: Only adding folders to stack with a compose file.

### LINKED ISSUE #183: Folders without compose config visible in UI
### ⚠️ Please verify that this bug has NOT been reported before.

- [X] I checked and didn't find similar issue

### 🛡️ Security Policy

- [X] I agree to have read this project [Security Policy](https://github.com/louislam/dockge/security/policy)

### Description

After #9 was fixed,  Dockge will now startup even if mounted directory has folders without compose config files. However, Dockge will still show these folders in the UI.

image

### 👟 Reproduction steps

1. Add a folder without a `compose.yaml` file in the the mounted directory.
2. Up dockge.
3. See the unwanted folder in the sidebar.

### 👀 Expected behavior

Not showing folders(/files) that are irrelevant for Dockge; if it does not contain a compose configuration file, it should not show up.

### 😓 Actual Behavior

image

### Dockge Version

1.1.1-nightly-20231125142720

### 💻 Operating System and Arch

Ubuntu 22.04.2 LTS

### 🌐 Browser

Google Chrome 119.0.6045.160 

### 🐋 Docker Version

24.0.5

### 🟩 NodeJS Version

_No response_

### 📝 Relevant log output

_No response_

### PR DESCRIPTION
**Pull request by Ozy-Viking**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/dockge/blob/master/CONTRIBUTING.md

Tick the checkbox if you understand [x]: 
- [x] I have read and understand the pull request rules.

# Description

Fixes #183

## Type of change

Please delete any options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- This change requires a documentation update (Maybe)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

### Before

image

### After

image

### CODE REVIEW

**Comment by Ozy-Viking** [comment]
It checks if the folder in the  stacks has any of the following files ` "compose.yaml", "compose.yml", "docker-compose.yml", "docker-compose.yaml" ` .

File: backend/stack.ts [hunk 1]
```diff
@@ -222,6 +222,26 @@ export class Stack {
...
 
+    /**
+     * Checks if a compose file exists in the specified directory.
+     * @async
+     * @static
+     * @param {string} stacksDir - The directory of the stack.
+     * @param {string} filename - The name of the directory to check for the compose file.
+     * @returns {Promise<boolean>} A promise that resolves to a boolean indicating whether any compose file exists.
+     */
+    static async composeFileExists(stacksDir : string, filename : string) : Promise<boolean> {
```
Comment (louislam) [inline backend/stack.ts:233]: I feel it is a bit over used await and promise here. I try to rewrite it

**Comment by louislam** [comment]
Actually I was thinking using something like `.gitignore` to exclude them, because file system operations will be fewer. 

I think it is ok at this moment.

**Comment by MrNavaStar** [comment]
> Actually I was thinking using something like `.gitignore` to exclude them, because file system operations will be fewer. 

I think it would be a good idea to also rely on the gitignore, but the gitignore often doesn't hide some main folders (like the .git folder) so if you are going this approach you would probably want something more custom.

File: backend/stack.ts (hunk 1 above)
Comment (Ozy-Viking) [reply backend/stack.ts:233]: No kidding, that is definitely a weak point for me hahaha

**Pull request merged**